qubesmanager/settings.py

```diff
@@ -664,27 +667,52 @@ def __init__(self, name, ident, parent=None):
         if self.vm.is_running():
             self.dev_list.setEnabled(False)
             self.turn_off_vm_to_modify_devs.setVisible(True)
+            self.no_strict_reset_button.setEnabled(False)
         else:
             self.dev_list.setEnabled(True)
             self.turn_off_vm_to_modify_devs.setVisible(False)
 
-
     def __apply_devices_tab__(self):
         msg = []
 
         try:
             old = [ass.ident.replace('_', ':')
-                for ass in self.vm.devices['pci'].persistent()]
+                   for ass in self.vm.devices['pci'].persistent()]
 
             new = [self.dev_list.selected_list.item(i).ident
-                    for i in range(self.dev_list.selected_list.count())]
+                   for i in range(self.dev_list.selected_list.count())]
             for ident in new:
                 if ident not in old:
+                    options = {}
+                    if ident in self.new_strict_reset_list:
+                        options['no-strict-reset'] = True
                     ass = devices.DeviceAssignment(
                         self.vm.app.domains['dom0'],
                         ident.replace(':', '_'),
-                        persistent=True)
+                        persistent=True, options=options)
                     self.vm.devices['pci'].attach(ass)
+                elif (ident in self.current_strict_reset_list) != \
+                        (ident in self.new_strict_reset_list):
+                    current_assignment = None
+                    for assignment in self.vm.devices['pci'].assignments(
+                            persistent=True):
+                        if assignment.ident.replace("_", ":") == ident:
+                            current_assignment = assignment
+                            break
+                    if current_assignment is None:
+                        # it would be very weird if this happened
```

Comment:
Yes, it would. Better log it as an error. Currently it _silently_ abort applying PCI device changes...